### PR TITLE
[Tokenizer] Add CLI Tokenizer Converter

### PR DESCRIPTION
**Pull request by apaniukov**
```bash
usage: convert_tokenizer [-h] [-o OUTPUT] [--with-detokenizer] [--skip_special_tokens] [--use-fast-false] [--trust-remote-code]
                         [--tokenizer-output-type {i32,i64}] [--detokenizer-input-type {i32,i64}] [--streaming-detokenizer STREAMING_DETOKENIZER]
                         name

Converts tokenizers from Huggingface Hub to OpenVINO Tokenizer model.

positional arguments:
  name                  The model id of a tokenizer hosted inside a model repo on huggingface.co or a path to a saved Huggingface tokenizer
                        directory

options:
  -h, --help            show this help message and exit
  -o OUTPUT, --output OUTPUT
                        Output directory
  --with-detokenizer    Add a detokenizer model to the output
  --skip_special_tokens
                        Produce detokenizer that will skip special tokens during decoding, similar to huggingface_tokenizer.decode(token_ids,
                        skip_special_tokens=True).
  --use-fast-false      Pass `use_fast=False` to `AutoTokenizer.from_pretrained`. It will initialize legacy HuggingFace tokenizer and then
                        converts it to OpenVINO. Might result in slightly different tokenizer. See models with _slow suffix https://github.com/op
                        envinotoolkit/openvino_contrib/tree/master/modules/custom_operations/user_ie_extensions/tokenizer/python#coverage-by-
                        model-type to check the potential difference between original and OpenVINO tokenizers
  --trust-remote-code   Pass `trust_remote_code=True` to `AutoTokenizer.from_pretrained`. It will execute code present on the Hub on your local
                        machine
  --tokenizer-output-type {i32,i64}
                        Type of the output tensors for tokenizer
  --detokenizer-input-type {i32,i64}
                        Type of the input tensor for detokenizer
  --streaming-detokenizer STREAMING_DETOKENIZER
                        [Experimental] Modify SentencePiece based detokenizer to keep spaces leading space. Can be used to stream a model output
                        without TextStreamer buffer

```

### CODE REVIEW

File: modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/constants.py [hunk 1]
```diff
@@ -1,3 +1,7 @@
+#!/usr/bin/env python3
```
Comment (ilya-lavrenov) [inline modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/constants.py:1]: do we need `!/usr/bin/env python3` ?

**Review by ilya-lavrenov — COMMENTED** [review]
maybe we need to integrate this package to HF and use optimum-cli instead of custom tool?

File: modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    args = get_parser().parse_args()
+    hf_tokenizer = AutoTokenizer.from_pretrained(args.name, trust_remote_code=args.trust_remote_code)
+    converted = convert_tokenizer(
+        hf_tokenizer, with_detokenizer=args.with_detokenizer, streaming_detokenizer=args.streaming_detokenizer
+    )
+    if not isinstance(converted, tuple):
+        converted = (converted,)
+
+    for converted_model, name in zip(converted, ("tokenizer", "detokenizer")):
```
Comment (slyalin) [inline modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py:77]: We are going to migrate this functionality under optimum-intel and export 3 models in optimum-cli, so please name the output model in a way that is aligned with the current openvino_model.xml name, so openvino_tokenizer.xml and openvino_detokenizer.xml.

File: modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        action="store_true",
+        help=(
+            "Pass `trust_remote_code=True` to `AutoTokenizer.from_pretrained`. It will "
+            "execute code present on the Hub on your local machine."
+        ),
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        required=False,
```
Comment (Wovchena) [inline modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py:51]: `args.verbose` is not used

File: modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+
+def convert_hf_tokenizer() -> None:
+    from transformers import AutoTokenizer
+
+    args = get_parser().parse_args()
+    hf_tokenizer = AutoTokenizer.from_pretrained(args.name, trust_remote_code=args.trust_remote_code)
+    converted = convert_tokenizer(
+        hf_tokenizer, with_detokenizer=args.with_detokenizer, streaming_detokenizer=args.streaming_detokenizer
+    )
```
Comment (Wovchena) [inline modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py:73]: I'm OK with the provided CLI options, but you may wat to expose the rest of them as well: `tokenizer_output_type` and `detokenizer_input_type`

**Review by Wovchena — APPROVED** [review]

File: modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/constants.py (hunk 1 above)
Comment (apaniukov) [reply modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/constants.py:1]: Changed to `# -*- coding: utf-8 -*-`

File: modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py (hunk 2 above)
Comment (apaniukov) [reply modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py:77]: Fixed.

File: modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py (hunk 4 above)
Comment (apaniukov) [reply modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py:73]: Added

File: modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py (hunk 3 above)
Comment (apaniukov) [reply modules/custom_operations/user_ie_extensions/tokenizer/python/openvino_tokenizers/cli.py:51]: Deleted.

**Review by ilya-lavrenov — APPROVED** [review]
let's also migrate to release branch

**Pull request merged**